### PR TITLE
Framework update

### PR DESCRIPTION
**Pull request by SSaishruthi**

### CODE REVIEW

File: api/predict.py [hunk 1]
```diff
@@ -1,60 +1,42 @@
-from flask_restplus import Namespace, Resource, fields
+from core.model import ModelWrapper
+from flask_restplus import fields
 from werkzeug.datastructures import FileStorage
+from werkzeug.exceptions import BadRequest
```
Comment (bdwyer2) [inline api/predict.py]: This can be deleted

File: api/predict.py [hunk 2]
```diff
@@ -1,60 +1,42 @@
...
 
     model_wrapper = ModelWrapper()
 
-    @api.doc('predict')
-    @api.expect(input_parser)
+    @MAX_API.doc('predict')
+    @MAX_API.expect(input_parser)
     def post(self):
         """Make a prediction given input data"""
         result = {'status': 'error'}
```
Comment (bdwyer2) [inline api/predict.py]: This can be deleted

File: core/model.py [hunk 3]
```diff
@@ -2,6 +2,8 @@
 import logging
 from config import DEFAULT_MODEL_PATH
 from PIL import Image
+from config import MODEL_META_DATA as model_meta
```
Comment (djalova) [inline core/model.py]: You can consolidate these imports from config.

File: api/predict.py [hunk 4]
```diff
@@ -7,54 +8,32 @@
 import os
```
Comment (djalova) [inline api/predict.py]: We should clean up the imports here. Reference: https://www.python.org/dev/peps/pep-0008/#imports

> Imports should be grouped in the following order:

> Standard library imports.
> Related third party imports.
> Local application/library specific imports.
> You should put a blank line between each group of imports.

File: api/predict.py (hunk 2 above)
Comment (djalova) [reply api/predict.py]: What did you mean here? We still need the
```
   @MAX_API.doc('predict')
    @MAX_API.expect(input_parser)
    @MAX_API.marshal_with(predict_response)
```

Comment (bdwyer2) [reply api/predict.py]: Just the `result = {'status': 'error'}`

File: api/predict.py (hunk 4 above)
Comment (SSaishruthi) [reply api/predict.py]: @djalova Modified predict.py.

 Is that ok?

Comment (djalova) [reply api/predict.py]: `os` should be at the top and `core.model` should be at the bottom in this case.

File: api/predict.py [hunk 5]
```diff
@@ -1,63 +1,44 @@
...
 
-    model_wrapper = ModelWrapper()
+class ModelPredictAPI(PredictAPI):
 
-    @api.doc('predict')
-    @api.expect(input_parser)
+    model_wrapper = ModelWrapper()
+    
+    @MAX_API.doc('predict')
+    @MAX_API.expect(input_parser)
```
Comment (djalova) [inline api/predict.py]: We still need the `@MAX_API.marshal_with(...)`

File: app.py [hunk 6]
```diff
@@ -1,14 +1,8 @@
...
-# load default config
-app.config.from_object('config')
-# load override config if exists
-if 'APP_CONFIG' in os.environ:
-    app.config.from_envvar('APP_CONFIG')
-api.init_app(app)
 
-if __name__ == '__main__':
-    app.run(host='0.0.0.0')
+max_app = MAXApp()
```
Comment (djalova) [inline app.py]: Need to pass in the API_TITLE, API_DESC, API_VERSION into `MAXApp()`

File: tests/test.py [hunk 7]
```diff
@@ -14,7 +14,7 @@ def test_swagger():
 
     json = r.json()
     assert 'swagger' in json
-    assert json.get('info') and json.get('info').get('title') == 'Model Asset Exchange Server'
+    assert json.get('info') and json.get('info').get('title') == 'Model Asset Exchange Microservice'
```
Comment (MLnick) [inline tests/test.py]: While we are here we should update:

* `API_TITLE` to be `MAX Image Completer`
* `API_DESC` to be the short description from https://developer.ibm.com/exchanges/models/all/max-image-completer/ - i.e. `Recognize and extract faces in an image and complete the corrupted portions.`
* `API_VERSION` accordingly (I guess `1.1.0`)

These should be put in `config.py` if not already there.

Tests should be updated accordingly

File: api/predict.py (hunk 5 above)
Comment (SSaishruthi) [reply api/predict.py]: It is throwing an error as we don't have predict_response.

Comment (djalova) [reply api/predict.py]: Oh right, ignore that then.

File: api/predict.py [hunk 8]
```diff
@@ -1,63 +1,44 @@
...
-        """Return the metadata associated with the model"""
-        return MODEL_META_DATA
+from maxfw.core import MAX_API, PredictAPI
+from api.pre_process import alignMain
+from core.model import ModelWrapper
 
 
 # Creating a JSON response model: https://flask-restplus.readthedocs.io/en/stable/marshalling.html#the-api-model-factory
-label_prediction = api.model('LabelPrediction', {
+label_prediction = MAX_API.model('LabelPrediction', {
```
Comment (MLnick) [inline api/predict.py]: If we don't need this we can remove it entirely

**Pull request merged**